src/batch/src/executor/table_function.rs

```diff
@@ -0,0 +1,313 @@
+
+            for _ in 0..chunk_size {
+                if cur > stop {
+                    break;
+                }
+                builder.append(Some(cur.as_scalar_ref())).unwrap();
+                cur = cur.checked_add(step.as_scalar_ref())?;
+            }
+
+            let arr = builder.finish()?;
+            let len = arr.len();
+            let columns = vec![Column::new(Arc::new(arr.into()))];
+            let chunk: DataChunk = DataChunk::new(columns, len);
+
+            yield chunk;
+        }
+    }
+}
+
+pub struct Unnest {
+    array_refs: Vec<ArrayRef>,
+    return_type: DataType,
+}
+
+impl Unnest {
+    #[try_stream(boxed, ok = DataChunk, error = RwError)]
+    async fn eval(self) {
+        let sum = self.array_refs.iter().map(|a| a.len()).sum();
+        let mut builder = self.return_type.create_array_builder(sum)?;
+        self.array_refs
+            .iter()
+            .try_for_each(|a| builder.append_array(a))?;
+        let arr = builder.finish()?;
+        let len = arr.len();
+        let columns = vec![Column::new(Arc::new(arr))];
+        let chunk: DataChunk = DataChunk::new(columns, len);
+
+        yield chunk;
+    }
+}
+
+pub struct TableFunctionExecutor {
+    schema: Schema,
+    identity: String,
+    table_function: TableFunction,
+}
+
+impl Executor for TableFunctionExecutor {
+    fn schema(&self) -> &Schema {
+        &self.schema
+    }
+
+    fn identity(&self) -> &str {
+        &self.identity
+    }
+
+    fn execute(self: Box<Self>) -> BoxedDataChunkStream {
+        match self.table_function {
+            TableFunction::GenerateSeriesTime(gs) => gs.eval(),
+            TableFunction::GenerateSeriesInt(gs) => gs.eval(),
+            TableFunction::Unnest(us) => us.eval(),
+        }
+    }
+}
+
+pub struct TableFunctionExecutorBuilder {}
+
+#[async_trait::async_trait]
+impl BoxedExecutorBuilder for TableFunctionExecutorBuilder {
+    async fn new_boxed_executor<C: BatchTaskContext>(
+        source: &ExecutorBuilder<C>,
+        inputs: Vec<BoxedExecutor>,
+    ) -> Result<BoxedExecutor> {
+        ensure!(
+            inputs.is_empty(),
+            "GenerateSeriesExecutor should not have child!"
+        );
+        let node = try_match_expand!(
+            source.plan_node().get_node_body().unwrap(),
+            NodeBody::TableFunction
+        )?;
+
+        let identity = source.plan_node().get_identity().clone();
+
+        let dummy_chunk = DataChunk::new_dummy(1);
+        let args: &Vec<ExprNode> = node.get_args();
+
+        let array_refs = args
+            .iter()
+            .map(|a| {
+                let expr = build_from_prost(a)?;
+                expr.eval(&dummy_chunk)
+            })
+            .collect::<Result<Vec<_>>>()?;
+        let return_type = DataType::from(node.data_type.as_ref().unwrap());
+        match node.get_series_type()? {
+            Generate => {
+                let start = array_refs[0].clone();
```

Comment:
pls split the branches into multiple functions:

```rust
impl GenerateSeriesTime {
  fn new(..) -> Result<GenerateSeriesTime> {
  }
}
```

And this function can be simplified as:

```rust
fn new_boxed_executor() {
  ...
  let table_func = match node.get_series_type()? {
    Generate => ...
    Unnest => Unnest::new()?
  };
  Ok(Box::new(TableExecutor { schema, identity, table_func }))  
}
```